### ISSUE TITLE
pytorch的loss 的backward不是会把所有相关参数的grads算好并存在.grad中吗？

### ISSUE THREAD

**Issue by egptee** [OpenLMLab/LOMO#6]
这样的话，grads的空间不是已经申请过了，这部分空间是怎么节省下来的呢？

**Comment by KaiLv69** [comment]
如果不定义优化器的话，p.grad在backward过程中应该是从全None根据反向传播顺序变为不是None的，显存占用逐渐增加。

**Comment by QipengGuo** [comment]
我们更新了一下Readme，里面简单介绍了一下反向过程。我们平时调用 loss.backward()后，所有的grads是计算好的，但它实际上是一步步计算出来的，参数的.grad属性也是按顺序赋值的。我们的方法是通过修改backward内部的计算流程来实现的。
可以参考PyTorch的官方文档来理解，https://pytorch.org/docs/stable/notes/autograd.html#backward-hooks-execution
